### PR TITLE
SOC-1443 Don't allow token-less GET requests to send a reply

### PR DESCRIPTION
**Pull request by garthwebb**
Note, I've based this off the SOC-1444 branch since it has some new functions I need.

https://wikia-inc.atlassian.net/browse/SOC-1443

### CODE REVIEW

File: extensions/wikia/Wall/WallExternalController.class.php [hunk 1]
```diff
@@ -613,12 +624,14 @@ public function replyToMessage() {
...
+		/** @var $wn WallNotifications */
 		$wn = new WallNotifications();
-		$wn->markRead( $this->wg->User->getId(), $this->wg->CityId, $this->request->getVal( 'parent' ) );
-
+		$wn->markRead(
+			$this->wg->User->getId(),
+			$this->wg->CityId,
+			$this->request->getVal( 'parent' )
+		);
+		gmark();
```
Comment (armonr) [inline extensions/wikia/Wall/WallExternalController.class.php]: Debug code

File: extensions/wikia/Wall/WallExternalController.class.php [hunk 2]
```diff
@@ -7,6 +7,8 @@
  */
 
 class WallExternalController extends WikiaController {
+	use \Wikia\Logger\Loggable;
```
Comment (armonr) [inline extensions/wikia/Wall/WallExternalController.class.php:10]: Is this used?

File: extensions/wikia/Wall/WallExternalController.class.php (hunk 1 above)
Comment (garthwebb) [inline extensions/wikia/Wall/WallExternalController.class.php]: Gah!

File: extensions/wikia/Wall/WallExternalController.class.php (hunk 2 above)
Comment (garthwebb) [inline extensions/wikia/Wall/WallExternalController.class.php:10]: Yah, you can see it here:
https://github.com/Wikia/app/pull/8853/files#diff-8546689cd7c7a5fca56d659448f6c2e2R601

**Comment by armonr** [comment]
:+1:

File: extensions/wikia/Wall/js/WallBackendBridge.js [hunk 3]
```diff
@@ -79,7 +79,8 @@
 					pagetitle: page.title,
 					pagenamespace: page.namespace,
 					convertToFormat: convertToFormat,
-					quotedFrom: quotedFrom || ''
+					quotedFrom: quotedFrom || '',
+					token: window.mw.user.tokens.get('editToken')
 				},
 				callback: this.proxy(function (data) {
```
Comment (kvas-damian) [inline extensions/wikia/Wall/js/WallBackendBridge.js:85]: Does this callback support error message returned when token is not valid?

Comment (garthwebb) [inline extensions/wikia/Wall/js/WallBackendBridge.js:85]: See my note on this in https://github.com/Wikia/app/pull/8854

**Comment by kvas-damian** [comment]
It's not optimal from UI perspective, but we're going to replace Forum and Wall by new Discussion product.

LGTM

**Pull request merged**